### PR TITLE
Init: don't re-checkout brave-core

### PR DESCRIPTION
**Pull request by petemill**
This is cleanup. No need to re-checkout brave-core since we won't be using gclient to manage git checkout for it anymore as of https://github.com/brave/brave-core/pull/11543. 

**Do not merge before that brave-core PR is uplifted to every current release branch.**

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A
  - [ ] https://github.com/brave/brave-browser/wiki/Brave-Wallet

### CODE REVIEW

File: scripts/init.js [hunk 1]
```diff
@@ -20,20 +20,8 @@ if (!fs.existsSync(path.join(braveCoreDir, '.git'))) {
   util.runGit(braveCoreDir, ['clone', util.getNPMConfig(['projects', 'brave-core', 'repository', 'url']), '.'])
   util.runGit(braveCoreDir, ['checkout', braveCoreRef])
```
Comment (atuchin-m) [inline scripts/init.js:21]: Do we really need this `init` script?
Now it does only mkdir + git clone brave-core + calling init.

Can `npm install` can simply clone brave-core?
And a user or CI can `git checkout <someRef>` after.

But maybe it's the next step;)

Comment (bridiver) [reply scripts/init.js:21]: we don't need it, but it's a nice helper to setup the directory structure and clone brave-core

**Review by bridiver — APPROVED** [review]

File: scripts/init.js (hunk 1 above)
Comment (bridiver) [reply scripts/init.js:21]: also for now it needs to be backwards compatible with CI

**Comment by atuchin-m** [comment]
++

File: scripts/init.js (hunk 1 above)
Comment (petemill) [reply scripts/init.js:21]: @atuchin-m yes the only reason for this repo is now to perform init. It's a good idea of yours to run `npm run init` as a `postinstall` script.

**Review by goodov — APPROVED** [review]

**Review by atuchin-m — APPROVED** [review]

**Comment by kjozwiak** [comment]
Adding missing `1.39.x` milestone as this landed on `master` which is currently on `1.39.x`.

**Pull request merged**